test/suites/test_push.py

```diff
@@ -0,0 +1,257 @@
+                    'on_push_ctx': None,
+                }
+            },
+            'output': {
+                'callback_res': [[[100, 1]], [[200, 1]], [[300, 1]]],
+                'resp': [1, 'Mike'],
+            },
+        },
+        'replace': {
+            'input': {
+                'args': ['tester', (1, 'Bill')],
+                'kwargs': {
+                    'on_push': push_callback,
+                    # on_push_ctx must be set manually when running the test.
+                    'on_push_ctx': None,
+                }
+            },
+            'output': {
+                'callback_res': [[[100, 1]], [[200, 1]], [[300, 1]]],
+                'resp': [1, 'Bill'],
+            },
+        },
+        'update': {
+            'input': {
+                'args': ['tester', 1],
+                'kwargs': {
+                    'op_list': [],
+                    'on_push': push_callback,
+                    # on_push_ctx must be set manually when running the test.
+                    'on_push_ctx': None,
+                }
+            },
+            'output': {
+                'callback_res': [[[100, 1]], [[200, 1]], [[300, 1]]],
+                'resp': [1, 'Bill'],
+            },
+        },
+        'upsert': {
+            'input': {
+                'args': ['tester', (1, 'Bill')],
+                'kwargs': {
+                    'op_list': [],
+                    'on_push': push_callback,
+                    # on_push_ctx must be set manually when running the test.
+                    'on_push_ctx': None,
+                }
+            },
+            'output': {
+                'callback_res': [[[100, 1]], [[200, 1]], [[300, 1]]],
+                # resp not used in the test output.
+                'resp': None,
+            },
+        },
+        'delete': {
+            'input': {
+                'args': ['tester', 1],
+                'kwargs': {
+                    'on_push': push_callback,
+                    # on_push_ctx must be set manually when running the test.
+                    'on_push_ctx': None,
+                }
+            },
+            'output': {
+                'callback_res': [[[100, 1]], [[200, 1]], [[300, 1]]],
+                'resp': [1, 'Bill'],
+            },
+        },
+    }
+
+    def test_00_00_push_via_connection(self):
+        for case_name in self.push_test_cases.keys():
+            with self.subTest(name=case_name):
+                callback_res = []
+                case = self.push_test_cases[case_name]
+                testing_function = getattr(self.conn, case_name)
+                case['input']['kwargs']['on_push_ctx'] = callback_res
+                resp = testing_function(
+                    *case['input']['args'],
+                    **case['input']['kwargs']
+                )
+                self.assertEqual(callback_res, case['output']['callback_res'])
+                if len(resp.data) > 0:
```

Comment:
I think it's better to rely on "test case resp is None" condition. Otherwise we may miss something

Reply:
Done, now using `case['output']['resp'] is not None` check